### PR TITLE
Spellcheck: ignore args to some latex commands

### PR DESCRIPTION
**Pull request by ShaneKilkelly**

### CODE REVIEW

File: public/coffee/ide/editor/directives/aceEditor/spell-check/SpellCheckManager.coffee [hunk 1]
```diff
@@ -53,6 +53,9 @@ define [
 			@$scope.learnWord = (highlight) =>
 				@learnWord(highlight)
 
+			# DEBUG
```
Comment (jpallen) [inline public/coffee/ide/editor/directives/aceEditor/spell-check/SpellCheckManager.coffee]: 🙃

File: public/coffee/ide/editor/directives/aceEditor/spell-check/SpellCheckManager.coffee [hunk 2]
```diff
@@ -246,3 +250,28 @@ define [
...
 			return $.ajax options
+
+		blacklistedCommandRegex: ///
+			\\                    # initial backslash
+			(label                # any of these commands
+			|[a-z]{0,2}ref
+			|usepackage
+			|begin
+			|end
+			|[a-z]{0,2}cite
```
Comment (jpallen) [inline public/coffee/ide/editor/directives/aceEditor/spell-check/SpellCheckManager.coffee]: There are `\...cite` commands with more than two prefix letters, e.g. `\parencite`. I'd imagine similarly for `\ref`. Is the concern about performance/scanning far ahead? If so, just something big like 16?

File: public/coffee/ide/editor/directives/aceEditor/spell-check/SpellCheckManager.coffee [hunk 3]
```diff
@@ -246,3 +250,28 @@ define [
...
+			|input
+			|include
+			|includegraphics)
+			( \[ [^\]]* \] )?     # optional [...] args
+				\{ [^}]*  \}        # the {...} args
+		///g
+
+		blankOutBlacklistedCommands: (line) ->
+			line.replace @blacklistedCommandRegex, (command) ->
+				command.replace(
```
Comment (jpallen) [inline public/coffee/ide/editor/directives/aceEditor/spell-check/SpellCheckManager.coffee]: Why do we need to do any analysis of the match, can we not just blank the whole thing? Am I right in saying this currently returns, e.g. `\usepackage[.........]{.............}`. Can we just return all `.............................` instead?

Comment (ShaneKilkelly) [inline public/coffee/ide/editor/directives/aceEditor/spell-check/SpellCheckManager.coffee]: Correct, and yes, the latter makes more sense :)

**Pull request merged**